src/settings/settingsdialog.cpp

```diff
@@ -165,6 +165,22 @@ SettingsDialog::~SettingsDialog()
 
 void SettingsDialog::accept()
 {
+    if (!ui->tesseractParametersTableWidget->validateParameters()) {
+        QMessageBox msgBox;
+        msgBox.setText(tr("The OCR parameter fields may not be empty."));
```

Comment:
Maybe is it more common to use "cannot" in this case? But I'm not a native English speaker, so just asking :)

```suggestion
        msgBox.setText(tr("The OCR parameter fields cannot be empty."));
```

Reply:
I looked it up. "May" implies permission to do something and "can" implies ability to do something. But must people just use them interchangeably and "can" is probably a bit more common. I'll change it to "can not" since "cannot" is an informal contraction.